django/db/backends/oracle/operations.py

```diff
@@ -310,17 +312,88 @@ def savepoint_create_sql(self, sid):
     def savepoint_rollback_sql(self, sid):
         return convert_unicode("ROLLBACK TO SAVEPOINT " + self.quote_name(sid))
 
+    @cached_property
+    def _foreign_key_constraints(self):
+        with self.connection.cursor() as cursor:
+            cursor.execute("""
+                SELECT
+                    cons.table_name,
+                    cons.constraint_name,
+                    rcons.table_name
+                FROM
+                    user_constraints cons
+                INNER JOIN
+                    user_constraints rcons ON cons.r_constraint_name = rcons.constraint_name
+                WHERE
+                    cons.constraint_type = 'R'
+            """)
+            return tuple(cursor.fetchall())
+
     def sql_flush(self, style, tables, sequences, allow_cascade=False):
-        # Return a list of 'TRUNCATE x;', 'TRUNCATE y;',
-        # 'TRUNCATE z;'... style SQL statements
         if tables:
-            # Oracle does support TRUNCATE, but it seems to get us into
-            # FK referential trouble, whereas DELETE FROM table works.
-            sql = ['%s %s %s;' % (
-                style.SQL_KEYWORD('DELETE'),
-                style.SQL_KEYWORD('FROM'),
-                style.SQL_FIELD(self.quote_name(table))
-            ) for table in tables]
+            constraints = []
+            truncated_tables = {table.upper() for table in tables}
```

Comment:
Assuming `tables` here comes from the apps, this will have a problem with long (>30 chars) table names. The Oracle metadata tables will hold the shortened names.